Python/bytecodes.c

```diff
@@ -1093,6 +1104,7 @@ dummy_func(
         inst(RETURN_VALUE, (retval -- res)) {
             assert(frame->owner != FRAME_OWNED_BY_INTERPRETER);
             _PyStackRef temp = retval;
+            assert(PyStackRef_IsHeapSafe(temp));
```

Comment:
Why the assert here? This blocks stackrefs from passing through to the host frame, which blocks a lot of unboxed and deferred refcounting optimizations.

Reply:
It blocks a lot of *unsafe* operations. Passing references towards the caller is unsafe as the only immediate 
(stored in `ob_refcnt`) reference  count to an object could be in the callee.